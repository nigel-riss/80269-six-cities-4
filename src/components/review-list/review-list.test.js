import React from 'react';
import renderer from 'react-test-renderer';
import ReviewList from './review-list.jsx';


const mock = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Max`,
    },
  },
  {
    comment: `Men are simple things. They can survive a whole weekend with only three things: beer, boxer shorts and batteries for the remote control.`,
    date: `2020-05-08T14:13:56.569Z`,
    id: 2,
    rating: 4.7,
    user: {
      avatarUrl: `img/avatar.svg`,
      id: 5,
      isPro: true,
      name: `Yuriy`,
    },
  },
  {
    comment: `I see you have something to talk about. Well, I have something to shout about. Infact something to sing about. But I'll just keep quiet and let you carry on.`,
    date: `2020-07-18T14:13:56.569Z`,
    id: 3,
    rating: 2.3,
    user: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: 6,
      isPro: true,
      name: `Angelina`,
    },
  },
];


describe(`ReviewList renders correctly`, () => {
  it(`without inNearPlaces passed`, () => {
    const tree = renderer
      .create(<ReviewList
        reviews={mock}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`for Property component`, () => {
    const tree = renderer
      .create(<ReviewList
        reviews={mock}
        isNearPlaces={true}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`for Main component`, () => {
    const tree = renderer
      .create(<ReviewList
        reviews={mock}
        isNearPlaces={false}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
