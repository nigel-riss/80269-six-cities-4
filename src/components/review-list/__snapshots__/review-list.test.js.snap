// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList renders correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-05-08T14:13:56.569Z"
      >
        May 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar.svg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Yuriy
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Men are simple things. They can survive a whole weekend with only three things: beer, boxer shorts and batteries for the remote control.
      </p>
      <time
        className="reviews__time"
        dateTime="2020-05-08T14:13:56.569Z"
      >
        May 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        I see you have something to talk about. Well, I have something to shout about. Infact something to sing about. But I'll just keep quiet and let you carry on.
      </p>
      <time
        className="reviews__time"
        dateTime="2020-07-18T14:13:56.569Z"
      >
        July 2020
      </time>
    </div>
  </li>
</ul>
`;
